src/modules/tools/reporter/Reporter.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+      This file is part of Smoothie (http://smoothieware.org/). The motion control part is heavily based on Grbl (https://github.com/simen/grbl).
+      Smoothie is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+      Smoothie is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+      You should have received a copy of the GNU General Public License along with Smoothie. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "libs/Module.h"
+#include "libs/Kernel.h"
+#include "Reporter.h"
+#include "Gcode.h"
+#include "StreamOutput.h"
+#include "TemperatureControlPublicAccess.h"
+#include "TemperatureControlPool.h"
+#include "TemperatureControl.h"
+#include "PublicData.h"
+#include "Robot.h"
+
+#include <math.h>
+#include <vector>
+
+
+Reporter::Reporter(){}
+
+Reporter::~Reporter(){}
+
+void Reporter::on_module_loaded(){
+    this->register_for_event(ON_GCODE_RECEIVED);
+}
+
+void Reporter::on_gcode_received(void *argument){
+    Gcode *gcode = static_cast<Gcode *>(argument);
+    if (gcode->has_m) {
+        if( gcode->m == 408 ) {
+            if (gcode->has_letter('S')) {
+                switch(static_cast<int>(gcode->get_value('S'))){
+                        case 0:
+                              // Beginning
+                              gcode->stream->printf("{\"status\":\"I\",");
+                              gcode->stream->printf("\"heaters{{\":[");
+                              for(auto it = THEKERNEL->temperature_control_pool->get_controllers().begin();
```

Comment:
This is not the correct way to keep a list of temperature controllers, there is a better way and it is used in several other modules. so you will need to change this to comply with the way it is currently done. It also wastes a lot of memory.